### PR TITLE
Update Build Instructions

### PR DESCRIPTION
**Pull request by basmaelgaabouri**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,181 @@
...
+- Intel SGX
+- Arm TrustZone
 
 ## Requirements
 
-The tests are done on a linux machine. [To enable docker experimental features on another OS.](https://docs.docker.com/engine/reference/commandline/checkpoint_create/)
+- **Docker:** 
+We use Docker to provide a consistent build environment.  Follow this guide to [install Docker](https://docs.docker.com/engine/install/) if necessary.
+- **Enable Docker squash experimental feature:** *The tests are done on a linux machine. [To enable docker experimental features on another OS.](https://docs.docker.com/engine/reference/commandline/checkpoint_create/)*
+We use `squash` docker experimental feature to help reduce the veracruz docker image size, to enable this feature:
```
Comment (dominic-mulligan-arm) [inline README.md:13]: We use the `squash`

Comment (dominic-mulligan-arm) [reply README.md:13]: Also, Veracruz should be capitalised.

File: README.md [hunk 2]
```diff
@@ -1,35 +1,181 @@
...
+    
+You can now start a shell in the newly created container:
+
+```
+docker exec -u <your username> -it 'veracruz' bash
+```
+
+## Test Instructions for SGX
+
+You can manual build veracruz by running:
```
Comment (dominic-mulligan-arm) [inline README.md:77]: sp: manually

**Comment by dominic-mulligan-arm** [comment]
Thanks for that, @basmaelgaabouri.  The changes look good barring a few typos here and there.

**Comment by dominic-mulligan-arm** [comment]
I did that process relatively recently (e.g. about a month ago) and the description sounds accurate (and actually more useful than the existing text, as the IAS doesn't actually give you anything that it refers to as a "token").

**Comment by dominic-mulligan-arm** [comment]
Though, FWIW, I don't think it matters if you register for an unsinkable or a linkable IAS account.

**Review by ShaleXIONG — APPROVED** [review]

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1+1=+1 reached, merging

**Pull request merged**